node/external/transactionAPI/unmarshaller.go

```diff
@@ -29,7 +33,13 @@ func (tu *txUnmarshaller) unmarshalReceipt(receiptBytes []byte) (*transaction.Ap
 		return nil, err
 	}
 
+	receiptHash, err := core.CalculateHash(tu.marshalizer, tu.hasher, rec)
```

Comment:
why do you have to calculate the receipt, since you fetch it from storage via a key? That key is the hash of the receipt. Am I missing anything?

Reply:
The storage key is, actually, the hash of the transaction that generated the receipt. However, moved the receipt hash computation to rosetta, since the "hash of a single receipt" is not meaningful on Node's side.